SimCalorimetry/EcalSimAlgos/interface/EcalSignalGenerator.h

```diff
@@ -81,6 +92,48 @@ class EcalSignalGenerator : public EcalBaseSignalGenerator {
     m_maxEneEB = (agc->getEBValue()) * theDefaultGains[1] * MAXADC * m_EBs25notCont;
     m_maxEneEE = (agc->getEEValue()) * theDefaultGains[1] * MAXADC * m_EEs25notCont;
 
+    if (m_timeDependent) {
+      //----
+      // Ecal LaserCorrection Constants for laser correction ratio
+      edm::ESHandle<EcalLaserDbService> laser;
+      eventSetup->get<EcalLaserDbRecord>().get(laser);
+
+      //
+      // FIXME: is this workaround of using "run" really needed or "time" can be used in MC generation as well?
+      //        check with generation experts
+      //
+      const edm::TimeValue_t eventTimeValue = theEvent->run();
+      //---- NB: this is a trick. Since the time dependent MC
+      //         will be based on "run" (and lumisection)
+      //         to identify the IOV.
+      //         The "time" defined here as "run"
+      //         will have to match in the generation of the tag
+      //         for the MC from ECAL (apd/pn, alpha, whatever time dependent is needed)
+      //
+      m_iTime = eventTimeValue;
+
+      m_lasercals = laser.product();
+
+      //
+      // the "prime" is exactly the same as the usual laser service, BUT
+      // it has only 1 IOV, so that effectively you are dividing IOV_n / IOV_0
+      // NB: in the creation of the tag make sure the "prime" (MC) tag is prepared properly!
+      // NB again: if many IOVs also in "MC" tag, then fancy things could be perfomed ... left for the future
+      //
+      edm::ESHandle<EcalLaserDbService> laser_prime;
+      eventSetup->get<EcalLaserDbRecordMC>().get(laser_prime);
+      //     const edm::TimeValue_t eventTimeValue = event.time().value();
+      m_lasercals_prime = laser_prime.product();
+
+      //clear the laser cache for each event time
+      //     CalibCache().swap(m_valueLCCache_LC);   -> strange way to clear a collection ... why was it like this?
```

Comment:
this use of `swap()` frees the memory associated with the container along with clearing the contents, whereas `clear()` is not required to free memory.